massdash/server/FeatureMapPlotterServer.py

```diff
@@ -0,0 +1,206 @@
+            if self.peak_picking_settings.do_peak_picking == 'Feature File Boundaries':
+                tr_group_feature_data = self.chrom_loader.loadTransitionGroupFeatures(self.transition_list_ui.transition_settings.selected_peptide,
+                    self.transition_list_ui.transition_settings.selected_charge)
+
+            elif self.peak_picking_settings.do_peak_picking in ['pyPeakPickerMRM', 'MRMTransitionGroupPicker', 'Conformer']:
+                # Perform peak picking if enabled
+                peak_picker = PeakPickingServer(self.peak_picking_settings, self.chrom_plot_settings)
+                tr_group_feature_data = peak_picker.perform_peak_picking(tr_group_data=chromatograms, 
+                                                                            spec_lib=self.chrom_loader.libraryFile)
+            elif self.peak_picking_settings.do_peak_picking == 'none':
+                tr_group_feature_data = None
+                pass
+
+            else:
+                raise ValueError(f"Invalid peak picking algorithm: {self.peak_picking_settings.do_peak_picking}. Valid options are 'Feature File Boundaries', 'pyPeakPickerMRM', 'MRMTransitionGroupPicker', and 'Conformer'.")
+
+            plot_settings_dict = self._get_plot_settings('Retention Time (s)', 'Intensity', 'chromatogram')
+            self._generate_plots_helper(chromatograms, plot_settings_dict, tr_group_feature_data)
+        
+        if self.chrom_plot_settings.display_mobilogram:
+            mobilograms = self.featureMapCollection.to_mobilograms()
+            plot_settings_dict = self._get_plot_settings('Ion Mobility (1/K0)', 'Intensity', 'mobilogram')
+            self._generate_plots_helper(mobilograms, plot_settings_dict)
+
+    def generate_two_dimensional_plots(self):
+        """
+        Generates two-dimensional plots based on the targeted data.
+
+        For each file in the targeted data, if the data is not empty, it retrieves the plot settings,
+        creates an instance of InteractiveTwoDimensionPlotter, generates the two-dimensional plots,
+        and stores them in the plot_obj_dict.
+
+        Returns:
+            None
+        """
+        for file, tr_df in self.targeted_data.items():
+            if not tr_df.empty():
+                plot_settings_dict = self._get_plot_settings(file)
+                plot_config = PlotConfig()
+                plot_config.update(plot_settings_dict)
+                plotter = InteractiveTwoDimensionPlotter(plot_config)
+                two_d_plots = plotter.plot(tr_df)
+                p = gridplot(two_d_plots, ncols=self.chrom_plot_settings.num_plot_columns, sizing_mode='stretch_width')
+                self.plot_obj_dict[file] = p
+        return self
+    
+    def generate_three_dimensional_plots(self):
+        """
+        Generates three-dimensional plots based on the targeted data.
+
+        Returns:
+            None
+        """
+        for file, tr_df in self.targeted_data.items():
```

Comment:
This throws the same error as the 2D case

![image](https://github.com/Roestlab/massdash/assets/32938975/be4e8c03-c6fe-4c80-be32-282916817ec1)